controller/setting_controller.go

```diff
@@ -259,69 +254,64 @@ func (sc *SettingController) syncBackupTarget() (err error) {
 	if err != nil {
 		return err
 	}
+	pollInterval := time.Duration(interval) * time.Second
 
-	if sc.bsMonitor != nil {
-		if sc.bsMonitor.backupTarget == targetSetting.Value &&
-			sc.bsMonitor.backupTargetCredentialSecret == secretSetting.Value {
-			// already monitoring
-			return nil
+	backupTarget, err := sc.ds.GetBackupTarget(defaultBackupTargetName)
+	if err != nil {
+		if !datastore.ErrorIsNotFound(err) {
+			return err
 		}
-		sc.logger.Infof("Restarting backup store monitor because backup target changed from %v to %v", sc.bsMonitor.backupTarget, targetSetting.Value)
-		sc.bsMonitor.Stop()
-		sc.bsMonitor = nil
-		manager.SyncVolumesLastBackupWithBackupVolumes(nil,
-			sc.ds.ListVolumes, sc.ds.GetVolume, sc.ds.UpdateVolumeStatus)
-	}
 
-	if targetSetting.Value == "" {
-		return nil
+		// Create the default BackupTarget CR if not present
+		backupTarget, err = sc.ds.CreateBackupTarget(&longhorn.BackupTarget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: defaultBackupTargetName,
+			},
+		})
 	}
+	// Update the default BackupTarget CR
+	backupTarget.Spec.BackupTargetURL = targetSetting.Value
+	backupTarget.Spec.CredentialSecret = secretSetting.Value
+	backupTarget.Spec.PollInterval = metav1.Duration{Duration: pollInterval}
 
-	target, err := manager.GenerateBackupTarget(sc.ds)
-	if err != nil {
+	if _, err = sc.ds.UpdateBackupTarget(backupTarget); err != nil && !datastore.ErrorIsConflict(err) {
 		return err
 	}
-	sc.bsMonitor = &BackupStoreMonitor{
-		logger:       sc.logger.WithField("component", "backup-store-monitor"),
-		controllerID: sc.controllerID,
-
-		backupTarget:                 targetSetting.Value,
-		backupTargetCredentialSecret: secretSetting.Value,
 
-		pollInterval: time.Duration(interval) * time.Second,
-
-		target: target,
-		ds:     sc.ds,
-		stopCh: make(chan struct{}),
+	if sc.bsTimer != nil {
+		if sc.bsTimer.backupTargetURL != targetSetting.Value ||
+			sc.bsTimer.backupTargetCredentialSecret != secretSetting.Value ||
+			sc.bsTimer.pollInterval != pollInterval {
+			// Stop backup store timer
+			sc.bsTimer.Stop()
+			sc.bsTimer = nil
+		}
 	}
```

Comment:
This is for better user experience if the user configures backup target URL and backup target credential secret at the same time. It restarts the timer and the timer sets BackupTarget CR `spec.syncRequestAt=time.Now()`.

Otherwise, if the user configures backup target URL and backup target credential secret at the same time, the setting_controller first sets backup target URL -> creates the timer -> sets backup target credential secret, the backup_target_controller be triggered after the poll interval time reached (default need to wait 5 mins).

Reply:
This is for better user experience if the user configures backup target URL and backup target credential secret at the same time. It restarts the timer and the timer sets BackupTarget CR `spec.syncRequestAt=time.Now()`.

Otherwise, if the user configures backup target URL and backup target credential secret at the same time, the setting_controller first sets backup target URL -> creates the timer -> sets backup target credential secret, the backup_target_controller be triggered after the poll interval time reached (default need to wait 5 mins).

Reply:
OK. It should be starting the timer as long as the backup target is **available** and the poll interval is set. And when the backup target spec is updated, Longhorn needs to set `spec.syncRequestedAt` immediately. Here, we don't need to care if the backupstore URL and the secret are set simultaneously.